src/goto-programs/remove_asm.cpp

```diff
@@ -281,20 +283,29 @@ void remove_asmt::process_instruction(
 void remove_asmt::process_function(
   goto_functionst::goto_functiont &goto_function)
 {
+  bool did_something=false;
+
   Forall_goto_program_instructions(it, goto_function.body)
   {
     if(it->is_other() && it->code.get_statement()==ID_asm)
     {
       goto_programt tmp_dest;
       process_instruction(*it, tmp_dest);
       it->make_skip();
+      did_something=true;
 
       goto_programt::targett next=it;
       next++;
 
       goto_function.body.destructive_insert(next, tmp_dest);
     }
   }
+
+  if(did_something)
+  {
+    remove_skip(goto_function.body);
```

Comment:
I think this would be unusual -- from what I've seen elsewhere, passes usually leave skips intact and then the driver program is expected to do one big `remove_skip` at a time of its choosing. This is especially useful in that it means the driver program can assume most passes do not alter location numbers.

Reply:
Doing `git grep remove_skip` we seem to have very much a mixed picture. I have opened #2067 to track this.